covid19-dashboard/dc-covid19/src/App.tsx

```diff
@@ -0,0 +1,363 @@
+import React from 'react';
+import OptionPanel from './OptionPanel'
+import SideNav from "./SideNav";
+import Row from "./Row";
+import numberWithCommas from "./NumberWithCommas";
+import moment from 'moment'
+import PanelInfo from './PanelInfo.json'
+
+type dataHolder = {
+    label: string,
+    value: number,
+    absolute: number
+}
+
+type casesAndDeathsHolder = {
+    cases: dataHolder[],
+    deaths: dataHolder[]
+}
+
+type stateType = {
+    allData: {},
+    region: string,
+    selectedDate: string,
+    availableDates: {
+        latest: string,
+        sevenDays: string,
+        fourteenDays: string,
+        thirtyDays: string,
+    }
+    availableRegions: {state: string, county: string},
+    showTopN: number,
+    daily: casesAndDeathsHolder,
+    dailyPerCapita: casesAndDeathsHolder,
+    dailyIncrease: casesAndDeathsHolder,
+    weekly: casesAndDeathsHolder,
+    weeklyPerCapita: casesAndDeathsHolder,
+    weeklyIncrease: casesAndDeathsHolder,
+    monthly: casesAndDeathsHolder,
+    monthlyPerCapita: casesAndDeathsHolder,
+    monthlyIncrease:  casesAndDeathsHolder,
+    absoluteCumulative: casesAndDeathsHolder,
+    cumulativePerCapita: casesAndDeathsHolder,
+    dcidMap: {},
+    animationClassName: string
+}
+
+let casesAndDeaths = {cases: [], deaths: []}
+class App extends React.Component <{}, stateType> {
+    constructor(props: {}) {
+        super(props);
+        // On-load fadein animation.
+        this.fadeInAnimation()
+        // Request latest data from server.
+        this.sendRequest().then(r => console.log("Data has been loaded!"))
+    }
+
+    state = {
+        allData: {}, // copy of all the unparsed data
+        region: "state", // current region selected
+        showTopN: 10, // number of top counties/states to show
+        selectedDate: "latest", // current date selected
+        availableRegions: {
+            state: "All States",
+            county: "All Counties"
+        },
+        availableDates: { // get the latest dates from server, these will be overwritten to the actual date once the server responds.
+            latest: "Latest",
+            sevenDays: "1 Week Ago",
+            fourteenDays: "2 Weeks Ago",
+            thirtyDays: "1 Month Ago",
+        },
+        daily: casesAndDeaths,
+        dailyPerCapita: casesAndDeaths,
```

Comment:
These variables are repeated in a few places. Imagine if someone wants to add a new chart. Can they just change one file and it would just work? Now it seems need to change quite some places.

Reply:
Resolved.